gulpfile.js

```diff
@@ -51,6 +52,49 @@ const { launch, connect } = require('hadouken-js-adapter');
 		console.log(`[${new Date().toLocaleTimeString()}] ${chalk[color][bgcolor](msg)}.`);
 	}
 
+	/** 
+	* Validates the current node version against supported node versions specified in this file
+	* Returns true of false if current node version is valid
+	* Currently only validates against a max node version
+	*
+	* Note: This method is being used instead of npm engines because of an npm bug where warnings don't print
+	* This bug was resolved in npm 6.12.0 but as that is a very new version of npm and is not linked to node 10.15.3
+	* in nvm we can't assume our users have access to this version.
+	*/
+	const isNodeVersionValid = () => {
+		let ret = true;
+		// split the current node version into major, minor and patch numbers
+		let version = process.version.split('v');
```

Comment:
It seems counterintuitive to me that we are validating the `MAX_NODE_VERSION` below, which is a constant string, but not validating `process.version`.  OTOH maybe Node guarantees that it will be of a certain format?  If I were you, I might validate this string by checking that it matches the regex `v([0-9])+\.([0-9])+\.([0-9])+`.  And then extract the three groups in parentheses.

Reply:
Hmm I'd figured process.version would always be consistent. I'll have to look into that. I'm only validating the max version because we/dev users can change it.